algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/workload.py

```diff
@@ -188,9 +185,31 @@ def loss_fn(
       mask_batch = torch.logical_and(mask_batch, target_lengths)
     else:
       mask_batch = target_lengths
-    n_valid_examples = max(mask_batch.sum(), 1)
+    n_valid_examples = mask_batch.sum().to(per_example_losses)
     summed_loss = per_example_losses.sum()
-    return summed_loss / n_valid_examples, per_example_losses
+    if USE_PYTORCH_DDP:
+      dist_nn.all_reduce(summed_loss)
```

Comment:
can you leave a comment here saying that we _need_ to use `dist_nn` in order to avoid problematic division by 8?

Reply:
Thank you, all fixed! 
Yes, len(ds) will not be expensive as it is implemented here: https://github.com/mlcommons/algorithmic-efficiency/blob/main/algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/libri_dataset.py#L31